fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/observer/eventmappings/AllNodeEventsOneEvent.java

```diff
@@ -70,59 +70,68 @@ public String apply(final Event ev) {
 
     @Override
     public Iterator<FedoraEvent> apply(final Iterator<Event> events) {
+        return new FedoraEventIterator(events);
+    }
 
-        return new Iterator<FedoraEvent>() {
+    private class FedoraEventIterator implements Iterator {
 
-            // sort JCR events into a Multimap keyed by the node ID involved
-            final Multimap<String, Event> sortedEvents = index(events, EXTRACT_NODE_ID);
+        final Iterator<Event> events;
 
-            final Iterator<String> nodeIds = sortedEvents.keySet().iterator();
+        public FedoraEventIterator(final Iterator<Event> events) {
```

Comment:
Please move the constructor below the following two instance variables:
`final Multimap<String, Event> sortedEvents;
 final Iterator<String> nodeIds;`